app/controllers/users_base_controller.rb

```diff
@@ -0,0 +1,6 @@
+class UsersBaseController < ApplicationController
+
+  def require_user
+    render 'public/404' unless current_user
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.